src/Parser/Rule/Common.idr

```diff
@@ -16,19 +16,19 @@ EmptyRule token ty = Grammar token False ty
 export
 location : {token : _} -> EmptyRule token (Int, Int)
 location
-    = do tok <- bounds peek
+    = do tok <- weakBounds peek
```

Comment:
Not sure what the point of `weakBounds` is here: we're extracting
the `Int`s anyway and throwing the `WithBounds` away?

Worth noting btw that:

* `(tok.startLine, tok.startCol)` can be refactored as `start tok`
* `(tok.endLine, tok.endCol)` can be refactored as `end tok`

Reply:
Yes, I introduced `weakBounds` because I didn't find a way to throw away position information in the existing code. If we use `bounds`, the position information will always stick around. That's the reason why some error messages wrap around to the next line, because the position information of the next token get merged with what has been just parsed.